### PR TITLE
 tuxfetch: init at 0.1.0

### PR DESCRIPTION
**Pull request by wiredhikari**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/tuxfetch/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "wiredhikari";
+    repo = pname;
+    rev = version;
+    sha256 = "1xlcnyy7zcnq78d9fv6k3vsawvbwzvailp21qskzsbpx299h3wpc";
+  };
+
+  cargoSha256 = "161nhazif24f7m555khmdvp9as2kndd47aisi8nbdwx91advyzkf";
+
+  meta = with lib; {
+    description = " A command-line system information tool written in rust";
```
Comment (fabaff) [inline pkgs/tools/misc/tuxfetch/default.nix:17]:
```suggestion
    description = "Command-line tool to get system information";
```

The "consumers" usually don't care about the programming language.

File: pkgs/tools/misc/tuxfetch/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://github.com/wiredhikari/tuxfetch";
+    license = licenses.unlicense;
```
Comment (fabaff) [inline pkgs/tools/misc/tuxfetch/default.nix:19]: The [LICENSE](https://github.com/wiredhikari/tuxfetch/blob/main/LICENSE) file says GPLv3.

```suggestion
    license = licenses.gpl3Only;
```

**Review by fabaff — COMMENTED** [review]
Entry in `all-packages.nix` is missing.

**Review by fabaff — COMMENTED** [review]
Please squash the commit of the package into one but keep the commit for `maintainers/maintainer-list.nix` separated.

Use `tuxfetch: init at 0.1.0` (according to the [guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes)).

**Review by fabaff — COMMENTED** [review]
The commit message for adding the package should be `tuxfetch: init at 0.1.0`.

**Comment by wiredhikari** [comment]
> The commit message for adding the package should be `tuxfetch: init at 0.1.0`.

Done

**Review by fabaff — COMMENTED** [review]
I don't get the system's information:

```bash
...ew/pr-173307-1]$ ./results/tuxfetch/bin/tuxfetch 
         .LINUS.
        $GENTOO$     ,"""""""""""""".
       @portage@|    | Linux Rules! |
       M|@||@) M|   _;..............'
       @,-----,@| -'
      C ^\__/^ rust
     aur        arch
    rpm          suse
   deb            enos
   MMM            MMMM
   MMM            MMMM
 __| ".        |\OS"vim
 |    `.       | `' \al
/     \.______.|     .'
\____  |Windows|   .'Imma
     `-'       `--' god
                       thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "Cursor position detection timed out." }', src/main.rs:18:43
                note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by fabaff** [comment]
On another host I get:

```bash
...ew/pr-173307]$ ./results/tuxfetch/bin/tuxfetch 
[...]
     aur        arch                        |UpTime    : 4d 4h 13m 
    rpm          suse                       |Terminal  : .nix-review-wra
   deb            enos                      |WM        : 
   MMM            MMMM                      |CPU       : Intel Core Processor (Skylake, IBRS)
thread 'main' panicked at 'index out of bounds: the len is 1 but the index is 2', /build/tuxfetch-0.1.0-vendor.tar.gz/nixinfo/src/lib.rs:73:17
 __| ".        |\OS"vim                                                                                                                       note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 |    `.       | `' \al
```

**Comment by wiredhikari** [comment]
Looking into it

**Pull request closed without merging**